src/main/java/org/springframework/data/couchbase/core/TemplateSupport.java

```diff
@@ -17,20 +17,28 @@
 
 import org.springframework.data.couchbase.core.mapping.CouchbaseDocument;
 import org.springframework.data.couchbase.core.mapping.event.CouchbaseMappingEvent;
+import org.springframework.data.couchbase.transactions.TransactionResultMap;
+
+import com.couchbase.transactions.TransactionGetResult;
 
 public interface TemplateSupport {
 
 	CouchbaseDocument encodeEntity(Object entityToEncode);
 
-	<T> T decodeEntity(String id, String source, long cas, Class<T> entityClass);
```

Comment:
If we consider that class internal I'm good with the changes, but if a user is consuming that we should not break the APIs.